src/utp_stream.cpp

```diff
@@ -2640,7 +2640,7 @@ bool utp_socket_impl::incoming_packet(span<std::uint8_t const> buf
 		m_reply_micro = timestamp - ph->timestamp_microseconds;
 		std::uint32_t prev_base = m_their_delay_hist.initialized() ? m_their_delay_hist.base() : 0;
 		their_delay = m_their_delay_hist.add_sample(m_reply_micro, step);
-		int base_change = m_their_delay_hist.base() - prev_base;
+		int base_change = int(m_their_delay_hist.base()) - int(prev_base); // TODO: review this sign conversion
```

Comment:
right, I don't think your change here is correct (and perhaps the original code isn't either). It's critical that the subtraction here is on *unsigned* types with wrapping semantics, and *then* cast to signed. the -10000 threshold seems a bit suspicious (as in, it should be a named constant). I imagine that's a threshold to avoid enabling this logic too eagerly.

please make ``base_change`` and ``prev_base`` const too (I believe they can be)